### PR TITLE
Product renewal url

### PR DESCRIPTION
**Pull request by sowmiyasubramaniyan**

### CODE REVIEW

File: app/Http/Controllers/HomeController.php [hunk 1]
```diff
@@ -457,4 +462,42 @@ private function getPHPCompatibleVersionString(string $version = null): string
...
+             $invoiceItems = $renewController->generateInvoice($product_details, $user, $order->id, $plan->id, $cost, $code = '', $item->agents, $oldcurrency);
+             $invoiceid = $invoiceItems->invoice_id;
+             $url = url("autopaynow/$invoiceid");
+
+             return $url;
+         } catch(\Exception $ex) {
+             $message = ['error' => $ex->getMessage()];
+
+             return response()->json($message);
+         }
```
Comment (ajaz-ul-haq) [inline app/Http/Controllers/HomeController.php:501]: Use models instead of DB statement and acoid unneceesary variables

**Review by ajaz-ul-haq — DISMISSED** [review]

File: app/Http/Controllers/HomeController.php [hunk 2]
```diff
@@ -457,4 +464,42 @@ private function getPHPCompatibleVersionString(string $version = null): string
     {
         return preg_replace('#v\.|v#', '', str_replace('_', '.', $version));
     }
+
+     public function renewurl(Request $request)
+     {
+         $validation = $request->validate([
+             'domain' => 'required|no_http',
```
Comment (RafficMohammed) [inline app/Http/Controllers/HomeController.php:471]: Create Request file

File: tests/Unit/HomeControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+class HomeControllerTest extends DBTestCase
+{
+    /**
+     * A basic unit test example.
+     *
+     * @return void
+     */
+    public function test_validation_when_given_url_empty()
+    {
+        $request = new Request(['domain' => 'example.com']);
```
Comment (RafficMohammed) [inline tests/Unit/HomeControllerTest.php:23]: remove unwanted code

**Review by RafficMohammed — CHANGES_REQUESTED** [review]
Fix that issue

**Review by RafficMohammed — APPROVED** [review]

**Review by SarfrazS — DISMISSED** [review]
Renewal API:- https://qa.productdemourl.com/saifbilling/public/renewurl?domain=sarfu.localhost/saiffaveo/public
image

Copy URL from response and paste on browser.
https://qa.productdemourl.com/saifbilling/public/autopaynow/13001
It is logged in then redirect to paynow page other with login then paynow page
image

**Pull request merged**